algorithmic.typ

```
@@ -9,6 +9,37 @@
  * (change_indent: int, body: ((ast | content)[] | content | ast)
  */
 
+#let locale = csv("locale.csv").fold((:), (result, row) => {
+  result + (row.at(0): row.slice(1))
+})
+#let locale = locale.pairs().fold((:), (result, (key, value)) => {
+  result+({key}: value.enumerate().fold((:), (result, (i, it)) => {
+    if it.len() != 0 {
+      result + (locale.at("en").at(i): it)
+    } else {
+      result
+    }
+  }))
+})
+#let l(kw) = context {
+  let lang = text.lang
+
+  if type(kw) == str {
+    let kws = locale.at(lang, default: locale.at("en"))
```

In which case, a default here is not necessary. It's better to throw an error if the corresponding lang does not exist.

---

Localization always falls back to default text. Using english isn't seen as wrong in most cases because even in LaTeX customizing algorithmx isn't all that common.

If we assume explicit `lang` argument approach, I'd like to be able to specify a language even if it isn't supported yet and for the document to compile. If the text isn't localized, then I'd switch to "en" if I don't want it to change in future, or keep "de" if I want the proper locales to be used once they're supported.

With using the current `text.lang` this is obviously the only reasonable approach.